### ISSUE TITLE
Upgrade to Node 12

### ISSUE THREAD

**Issue by kriszyp** [amapili/weak-value-map#10]
There are a number of breaking changes in the V8 API that require updates to compile for Node 12.

**Comment by mvduin** [comment]
After hitting the compile errors in Node 12, I remember I'd been fiddling with this module a while ago to try to use the "proper" context-aware APIs to the best of my ability, and [fix some other issues](https://github.com/dutchanddutch/weak-value-map/commit/593aaa06628a). I found that work again sitting on my laptop and with some minor fixes it seems to work in both node 10 and 12.

In case it's of interest: https://github.com/dutchanddutch/weak-value-map

For the most part this is just my best guess at what the Right Thing is to do, since API documentation seems to be pretty non-existant. My branch also has some purely cosmetic changes, like the stuff in utility.h. I have not performed any testing yet beyond "npm test".

**Comment by kriszyp** [comment]
@mvduin based on the discussion in https://github.com/amapili/weak-value-map/pull/9 it sounded like this wasn't complete, so I had taken the specific changes needed for upgrading the V8 APIs and put them in https://github.com/amapili/weak-value-map/pull/11. It looked like there some nice changes in https://github.com/amapili/weak-value-map/pull/9 if it were completed.

**Comment by mvduin** [comment]
Honestly, I just don't remember what I was doing before other things pulled me away and I ended up kind of forgetting about all this. It looks like I asked for #9 to be closed just because I messed up the pull request (by doing a pull request from my master branch and then pushing more changes to it, instead of creating a new branch to pull) rather than because there was anything wrong with it per se.

Although my master branch could still be considered work in progress, that's probably mainly because I have tendencies towards perfectionism and am seldom truly happy with my code. Nevertheless, my impression is that it's in a better state than the master branch here.
